### ISSUE TITLE
CKEditor : is disabled and data is clear after minimizing the dialog 

### ISSUE THREAD

**Issue by geekdiv** [primefaces-extensions/primefaces-extensions.github.com#522]
I have `pe:ckEditor`  inside my dialog. when I minimize and restore the dialog again, the data in editor is gone and editor is disabled. 

I have tried both primefaces editor and primefaces ckEditor but facing same issue on both.

## 1) Environment

- primefaces-6.1

- It does not work on Primeface showcase as well

- primefaces-extensions-6.1.1

- jsf-2.2

## 2) Expected behavior

After restore dialog editor data not clear and editor should not disable.

## 3) Actual behavior

After restore dialog editor data is clear and editor is disable. Instead of editor data display 'true' like below screenshot (Using Firefox)
![image](https://i.stack.imgur.com/SMIqc.png)

## 4) Steps to reproduce

Open Dialog add some data in editor and minimize dialog then again maximize that dialog.

## 5) Sample XHTML
```
 <p:dialog minimizable="true" modal="false" id="newDlg" widgetVar="dlgNew">
        	<p:editor value="test"/>
 </p:dialog>
```
```
<p:dialog closeOnEscape="true" responsive="true" widgetVar="dlg" id="dlg1" resizable="false" draggable="true" closable="true" dynamic="true" modal="false"  width="97%" minimizable="true" appendTo="@(body)" header="Dialog" onShow="PF('dlg').initPosition()">
   <pe:ckEditor toolbar="#{config['ckEditor.toolBar']}" id="body" widgetVar="editor" readonly="false" width="100%" maxlength="50000" value="#{myBean.textVal}" />
</p:dialog>
```
Note : suggested by @tandraschko  
https://github.com/primefaces/primefaces/issues/3206

**Comment by melloware** [comment]
@jxmai can you take a look at this one?

**Comment by jxmai** [comment]
Alirght, I will investigate

**Comment by jxmai** [comment]
@melloware I roughly understand what's going on, but I would like your opinion about how to fix it.

---

**Test code**:


```xhtml
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:ui="http://java.sun.com/jsf/facelets"
      xmlns:f="http://java.sun.com/jsf/core"
      xmlns:p="http://primefaces.org/ui"
      xmlns:pe="http://primefaces.org/ui/extensions"
      xmlns:h="http://java.sun.com/jsf/html">

    <h:head>   
        <title>PrimeFaces Test</title>
    </h:head>
    <h:body>
        
        <h:form>
			<p:commandButton value="open dialog" oncomplete="PF('dlgNew').show()" process="@this"></p:commandButton>
		</h:form>
		
		 <p:dialog minimizable="true" modal="false" id="newDlg" widgetVar="dlgNew" >
			<pe:ckEditor value="test"></pe:ckEditor>
			<p:textEditor value="test"></p:textEditor>
			<!-- <p:editor value="test"/> -->
		 </p:dialog>
    </h:body>
</html>
```


**Steps to replicate**

1. put a break point after dockingZone is finished initializing.

```js
dockingZone = $(document.body).children('.ui-dialog-docking-zone');
```

2. invoke the following script in debugger

```js
this.jq.appendTo(dockingZone);
```

then CKEditor will be broken at this stage

![image](https://user-images.githubusercontent.com/10467831/35344083-dc39d902-00f1-11e8-952f-9150802bc5d4.png)

---


The reason is  mentioned in this post: https://stackoverflow.com/questions/18275736/ckeditor-broken-after-remove-and-append-to-dom

> The logic behind CKEditor instance relies on DOM structure so much that it will, indeed, get broken, if you modify it.


The solution won't work for us elegantly since it requires the dialog has knowledge of what's inside.

**Comment by tandraschko** [comment]
Maybe we should also try to reimplement the minimize itself to just display:none the dialog and some placeholder to the docking zone.
I think many widgets will be broken if you minimize

**Comment by jxmai** [comment]
I agree :+1: . The dialog minimize and remove minimizing currently is appending dom back and forth which is not a good design. For this problem the design is a dead end for us.

**Comment by tandraschko** [comment]
Created a issue in PF. A PR for 6.3 is very welcome.

**Comment by tandraschko** [comment]
Will close this one, we can't really fix it here.

**Comment by melloware** [comment]
Fixed in PF 14.0.0: https://github.com/primefaces/primefaces/pull/11216
